crates/matrix-sdk/src/room_preview.rs

```diff
@@ -372,3 +375,92 @@ async fn search_for_room_preview_in_room_directory(
 
     Ok(None)
 }
+
+// Make sure the server name of the room id/alias is
+// included in the list of server names to send if no server names are provided
+fn ensure_server_names_is_not_empty(
+    own_server_name: Option<&ServerName>,
+    server_names: Vec<OwnedServerName>,
+    room_or_alias_id: &RoomOrAliasId,
+) -> Vec<OwnedServerName> {
+    let Some(own_server_name) = own_server_name else { return server_names };
+    if server_names.is_empty() {
+        if let Some(server_name) = room_or_alias_id.server_name() {
+            if server_name != own_server_name {
+                return server_names.into_iter().chain(vec![server_name.to_owned()]).collect();
```

Comment:
You could have just pushed the additional name to the vec and avoid the early return.

```suggestion
                server_names.push(server_name.to_owned());
```

Reply:
`server_names` is not mutable. I can make it so, but then I'd be creating a new `Vec` even when no changes are needed.

Reply:
But you wouldn't, you would return the same `Vec` that was passed to the method, some times you would mutate it and sometimes you won't.

The `collect()` creates a brand new `Vec`.

Reply:
Oh, I completely misunderstood how this works then (you'd actually be creating a different type on Kotlin, i.e.). I'll change that, thanks!